store/postgres/src/relational_queries.rs

```diff
@@ -586,6 +558,48 @@ impl EntityData {
     }
 }
 
+#[derive(QueryableByName, Clone, Debug, Default, Eq)]
+pub struct EntityDataExt {
+    #[diesel(sql_type = Text)]
+    pub entity: String,
+    #[diesel(sql_type = Jsonb)]
+    pub data: serde_json::Value,
+    #[diesel(sql_type = Integer)]
+    pub block_number: i32,
+    #[diesel(sql_type = Text)]
+    pub id: String,
+    #[diesel(sql_type = BigInt)]
+    pub vid: i64,
+}
+
+impl Ord for EntityDataExt {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let ord = self.block_number.cmp(&other.block_number);
+        if ord != Ordering::Equal {
+            ord
+        } else {
+            let ord = self.entity.cmp(&other.entity);
+            if ord != Ordering::Equal {
+                ord
+            } else {
+                self.id.cmp(&other.id)
+            }
+        }
+    }
+}
```

Comment:
One thing that confuses me about the sort order `(block_number, entity type, entity id)` is that entries for different entities can be interleaved, like if there are two entities `a` and `b`, you might end up with `[ (2, a), (2,b), (3, a), (3, b)]` and since `find_range` processes the entries in the vecs in order, can't you end up comparing an entry for `a` with one for `b`?

Reply:
You have to keep in mind that matching is always done between elements in the different vectors, and never between elements in the same one. Also a missing element in `upper_vec` is a `creation` and missing element in `lower_vec` is `deletion`, where a match (a presence of the equivalent respective values in both vectors) is a `modification`. And finally the two pairs of cases where there is either no matching element on one side or no remaining element at all on that same side, are in essence equivalent and produce the same result. Advancements of iterators are done on the already evaluated element(s). 

Probably a simple example would be more explanatory. 
Imagine you have elements in `lower_vec` like this: 
`[ (2, a), (2, c), (3, a) (3, b), (3, c) ]`, 

and in the `upper_vec` something like: 
`[ (2, a), (2, b), (3, a), (3, c), (4, a) ]`,

the strategy will first find a match of `(2, a)` in both vectors so modification of `a` in block `2`,  then `(2, b) `is missing in `lower_vec` so deletion of `b` in `2`, then `(2, c)` is missed in `upper_vec` so creation there, then `(3, a)` matched again so another modification, `(3, b)` only in `lower_vec` - creation, `(3, c)` on both sides - modification, `(4, a)` only in `upper_vec` - deletion.

Looking again at the comments at the start of the big while loop I tried to enter the same content as the first paragraph here, but also referred to some other aspects and that might had made it less readable. Should I try to improve it?